core/src/main/scala/kafka/log/FileMessageSet.scala

```diff
@@ -157,49 +157,36 @@ class FileMessageSet private[kafka](@volatile var file: File,
   /**
    * Search forward for the message whose timestamp is greater than or equals to the target timestamp.
    *
-   * The search will stop immediately when it sees a message in format version before 0.10.0. This is to avoid
-   * scanning the entire log when all the messages are still in old format.
-   *
    * @param targetTimestamp The timestamp to search for.
    * @param startingPosition The starting position to search.
-   * @return None, if no message exists at or after the starting position.
-   *         Some(the_next_offset_to_read) otherwise.
+   * @return The timestamp and offset of the message found. None, if no message is found.
    */
-  def searchForTimestamp(targetTimestamp: Long, startingPosition: Int): Option[Long] = {
+  def searchForTimestamp(targetTimestamp: Long, startingPosition: Int): Option[TimestampOffset] = {
     var maxTimestampChecked = Message.NoTimestamp
     var lastOffsetChecked = -1L
-    val messagesToSearch = read(startingPosition, sizeInBytes)
+    val messagesToSearch = read(startingPosition, sizeInBytes())
```

Comment:
The warning is probably because the method definition has parenthesis. Maybe we can just remove them and keep the old usage?